### ISSUE TITLE
riot tag based coverage report incorrectness

### ISSUE THREAD

**Issue by deeperx** [gotwarlost/istanbul#571]
Can't generate valid coverage reports from compiled tag files. For example: With karma lcov reporting, compiled tag file's coverage is not correctly reported.

# Reproduce:

**Example project:**

git clone https://github.com/vitogit/tdd-mocha-chai-riot

**Karma Config (_updated_):**

```
module.exports = function(config) {
  config.set({
    basePath: '',
    frameworks: ['mocha','chai','riot'],
    plugins: [
      'karma-mocha',
      'karma-mocha-reporter',
      'karma-phantomjs-launcher',
      'karma-chai',
      'karma-riot',
      'karma-coverage'
    ],
    files: [
      'node_modules/chai/chai.js',
      'src/**/*.tag',
      'test/**/*.js'
    ],
    preprocessors: {
      '**/*.tag': ['riot', 'coverage'],
      'src/**/*.js': ['coverage']
    },
    coverageReporter: {
      dir: 'coverage/',
      reporters: [
        { type: 'html', subdir: 'report-html' },
        { type: 'text', subdir: '.', file: 'text.txt' },
        { type: 'text-summary', subdir: '.', file: 'text-summary.txt' },
        { type: 'lcovonly', subdir: '.', file: 'lcov.info' }
      ]
    },
    browsers: ['PhantomJS'],
    reporters: ['mocha', 'coverage'],
    failOnEmptyTestSuite: false,
    autoWatch: true
  })
}
```

**How to run?**

```
npm i && npm i karma-coverage --save-dev
karma start
cat coverage/lcov.info
```

**_You can also see html based coverage reports inside coverage directory's html coverage reports._**

**LCOV is incorrect:**

<img width="711" alt="screen shot 2016-03-14 at 15 10 26" src="https://cloud.githubusercontent.com/assets/621232/13745487/f897271c-e9f6-11e5-8df4-aaa62413dd7d.png">

**HTML is correct:**

<img width="612" alt="screen shot 2016-03-14 at 15 11 51" src="https://cloud.githubusercontent.com/assets/621232/13745513/1ab6fa8e-e9f7-11e5-9b29-d8265d4c3230.png">

**Comment by deeperx** [comment]
By the way, actually, what I really wonder is why html coverage output is true while lcov coverage output is not?

**Comment by weblogixx** [comment]
I have the same problem. Lcov-Reporter shows incorrect line numbers, which makes me guess it really works on the already transpiled files (in this case react, but I do not think that this matters). 

Output for text, html and cobertura seem to be fine... The main problem is that I cannot get other plugins to work in SonarQube, so I have to use lcov...

Is there any news on this problem?

**Comment by deeperx** [comment]
@weblogixx Karma made everything complex. So, I've switched to more _basic_ approach: Run mocha and istanbul directly from CLI through npm run. This way, what happens is clear and LCOV also works correctly, because I don't need to deal with karma plugins etc. I'm directly dealing with the core components directly.

**Comment by weblogixx** [comment]
@deeperx Thank you for your answer. This is interesting, it seems to be more like a karma coverage reporter problem then? Unfortunately, I am not able to change my setup from karma to pure mocha (I know it technically is possible, I already did it in one or two projects I am currently working on, but I have some projects that really depend on karma :().

**Comment by deeperx** [comment]
Yes, it seems like that. It made so mad that I'd completely reconfigured to pure CLI based tools :)

Inanc Gumus

21 Tem 2016 tarihinde 08:23 saatinde, Chris <notifications@github.com> şunları yazdı:
